src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -149,13 +156,16 @@ public function onSubmit(FormEvent $event)
         }
 
         if ($this->deleteEmpty) {
-            $previousData = $event->getForm()->getData();
+            $previousData = $form->getData();
+            $entryFilter = $this->entryFilter;
             foreach ($form as $name => $child) {
+                /* @var $child FormInterface */
                 $isNew = !isset($previousData[$name]);
+                $needRemove = $child->isEmpty() || (null !== $entryFilter && !$entryFilter($child->getData()));
```

Comment:
It is about whether the child form/data is empty or not, so the previous name `isEmpty` sounds better IMO ;)